lib/Dialect/SystemC/SystemCOps.cpp

```diff
@@ -33,11 +35,29 @@ static void printImplicitSSAName(OpAsmPrinter &p, Operation *op,
 // SCModuleOp
 //===----------------------------------------------------------------------===//
 
-void SCModuleOp::getPortsOfDirection(PortDirection direction,
-                                     SmallVector<Value> &outputs) {
+static hw::PortDirection getDirection(Type type) {
+  return TypeSwitch<Type, hw::PortDirection>(type)
+      .Case<InOutType>([](auto ty) { return hw::PortDirection::INOUT; })
+      .Case<InputType>([](auto ty) { return hw::PortDirection::INPUT; })
+      .Case<OutputType>([](auto ty) { return hw::PortDirection::OUTPUT; });
```

Comment:
Should this defensively code a default with an assert?

Reply:
`TypeSwitch` already has an assert when no case matches. The enclosing function is part of the stack trace and thus easy to find.

Reply:
Oh, TIL!  FWIW it appears this is only the behavior when accessing the return value-- using the void specialization (`TypeSwitch<Type>`) or even using `TypeSwitch<Type,int>` won't assert on missed cases if you ignore the return value.
This seems reasonable, certainly for the void case, but good to know.  Anyway, thanks!